9-regular-expressions/01-regexp-introduction/article.md

```diff
@@ -1,91 +1,91 @@
-# Patterns and flags
+# الگو ها و پرچم ها (Patterns and flags)
 
-Regular expressions are patterns that provide a powerful way to search and replace in text.
+عبارات باقاعده (Regular expressions) الگوهایی هستند که روشی قدرتمند برای جستجو و جایگزینی در متن ارائه می دهند.
 
-In JavaScript, they are available via the [RegExp](mdn:js/RegExp) object, as well as being integrated in methods of strings.
+در جاوا اسکریپت، آنها از طریق شیء [RegExp](mdn:js/RegExp) استفاده می شوند و همچنین به خوبی با متد های رشته ها ادغام می شوند.
 
-## Regular Expressions
+## عبارات باقاعده (Regular Expressions)
 
-A regular expression (also "regexp", or just "reg") consists of a *pattern* and optional *flags*.
+عبارت باقاعده (همچنین "regexp"، یا فقط "reg") از یک *الگو(pattern)* و *پرچم(flags)* های اختیاری تشکیل می شوند.
 
-There are two syntaxes that can be used to create a regular expression object.
+دو روش وجود دارد که می توان از آنها برای ایجاد یک شیء عبارت باقاعده استفاده کرد.
 
-The "long" syntax:
+سینتکس "طولانی":
 
 ```js
-regexp = new RegExp("pattern", "flags");
+regexp = new RegExp("pattern(الگو)", "flags(پرچم)");
 ```
 
-And the "short" one, using slashes `"/"`:
+و سینتکس "کوتاه" نیز، استفاده از اسلش `"/"` می باشد.
 
 ```js
-regexp = /pattern/; // no flags
-regexp = /pattern/gmi; // with flags g,m and i (to be covered soon)
+regexp = /pattern/; // بدون پرچم
+regexp = /pattern/gmi; // با پرچم g, m, i (به زودی پوشش داده می شود)
 ```
 
-Slashes `pattern:/.../` tell JavaScript that we are creating a regular expression. They play the same role as quotes for strings.
+اسلش های `pattern:/.../` به جاوا اسکریپت می گوید که ما در حال ایجاد یک عبارت باقاعده هستیم. آنها همان نقش کوتیشن(') را برای رشته ها بازی می کنند.
 
-In both cases `regexp` becomes an instance of the built-in `RegExp` class.
+در هر دو مورد، `regexp` به یک نمونه از کلاس داخلی `RegExp` تبدیل می‌ شود.
 
-The main difference between these two syntaxes is that pattern using slashes `/.../` does not allow for expressions to be inserted (like string template literals with `${...}`). They are fully static.
+تفاوت اصلی بین این دو سینتکس این است که الگو با استفاده از اسلش‌های `/.../`، اجازه درج عبارات را نمی دهد (مانند حروف الفبای قالب(template literals) رشته با `${...}`). که آنها کاملا ساکن هستند.
 
-Slashes are used when we know the regular expression at the code writing time -- and that's the most common situation. While `new RegExp` is more often used when we need to create a regexp "on the fly" from a dynamically generated string. For instance:
+اسلش ها زمانی استفاده می شوند که عبارت باقاعده را در زمان نوشتن کد بدانیم -- و این رایج ترین حالت است. در حالی که `new RegExp` بیشتر زمانی استفاده می شود که ما نیاز به ایجاد یک regexp "در حال پرواز" از یک رشته تولید شده به صورت پویا داشته باشیم. برای مثال:
 
 ```js
-let tag = prompt("What tag do you want to find?", "h2");
+let tag = prompt("چه بر چسبی می خواهید پیدا کنید؟", "h2");
```

Comment:
```suggestion
let tag = prompt("کدام تگ HTML را می خواهید پیدا کنید؟", "h2");
```
Since this string will be shown in the browser, it'll have RTL format and will be readable to the user.